service/tasks/driver.py

```diff
@@ -1132,8 +1132,33 @@ def add_floating_ip(driverCls, provider, identity, core_identity_uuid,
                 floating_ip_addr, instance_alias)
             floating_ip = selected_floating_ip
         else:
-            floating_ip = network_driver.associate_floating_ip(instance_alias)
-            floating_ip_addr = floating_ip["floating_ip_address"]
+            if core_identity.provider.cloud_config['network']['topology'] \
+                    == "External Router Topology":
+                # Determine correct external network based on external gateway
+                # info of the identity's public router
+                public_router_name = core_identity.get_credential('router_name')
```

Comment:
I meant to say core_identity earlier in my previous comment. If you do a quick search in the codebase the credentials object is never accessed off an rtwo identity, its a copy of the credental set model on the core identity.